services/src/handlers/workflows.rs

```diff
@@ -735,4 +1036,117 @@ mod tests {
+                        dataset: dataset.clone(),
+                    },
+                }
+                .boxed(),
+            ),
+        };
+
+        let workflow_id = ctx
+            .workflow_registry_ref_mut()
+            .await
+            .register(workflow)
+            .await
+            .unwrap();
+
+        // create dataset from workflow
+        let res = warp::test::request()
+            .method("POST")
+            .path(&format!("/datasetFromWorkflow/{}", workflow_id.to_string(),))
+            .header(
+                "Authorization",
+                format!("Bearer {}", session_id.to_string()),
+            )
+            .body(
+                r#"{
+                "name": "foo",
+                "description": null,
+                "query": {
+                    "spatial_bounds": {
+                        "upperLeftCoordinate": {
+                            "x": -10.0,
+                            "y": 80.0
+                        },
+                        "lowerRightCoordinate": {
+                            "x": 50.0,
+                            "y": 20.0
+                        }
+                    },
+                    "time_interval": {
+                        "start": 1388534400000,
+                        "end": 1388534401000
+                    },
+                    "spatial_resolution": {
+                        "x": 0.1,
+                        "y": 0.1
+                    }
+                }
+            }"#,
+            )
+            .reply(&dataset_from_workflow_handler(ctx.clone()))
+            .await;
+
+        assert_eq!(res.status(), 200);
+
+        let body = std::str::from_utf8(res.body()).unwrap();
+        let response = serde_json::from_str::<RasterDatasetFromWorkflowResult>(body).unwrap();
+        // automatically deletes uploads on drop
+        let _test_uploads = TestDataUploads {
+            uploads: vec![response.upload],
+        };
+
+        // query the newly created dataset
+        let op = GdalSource {
+            params: GdalSourceParameters {
+                dataset: response.dataset.clone(),
+            },
+        }
+        .boxed();
+
+        let session = ctx.default_session_ref().await.clone();
+        let exe_ctx = ctx.execution_context(session).unwrap();
+
+        let o = op.initialize(&exe_ctx).await.unwrap();
+
+        let query_ctx = ctx.query_context().unwrap();
+        let query_rect = RasterQueryRectangle {
+            spatial_bounds: SpatialPartition2D::new((-10., 80.).into(), (50., 20.).into()).unwrap(),
+            time_interval: TimeInterval::new_unchecked(1_388_534_400_000, 1_388_534_400_000 + 1000),
+            spatial_resolution: SpatialResolution::zero_point_one(),
+        };
+
+        let processor = o.query_processor().unwrap().get_u8().unwrap();
+
+        let result = raster_stream_to_geotiff_bytes(
+            processor,
+            query_rect,
+            query_ctx,
+            Some(0.), // TODO
```

Comment:
I guess we missed an unclear TODO, here

Reply:
it was redundant. I removed it e0b69d551c29cacacefef9cc9b7b06fd78404fab